modules/GeneticsPortal.py

```diff
@@ -0,0 +1,534 @@
+
+        # Adding reported trait:
+        return_value['reported_trait'] = self.reported_trait
+
+        # The literature link is only added if the pubmed ID is present:
+        return_value['unique_experiment_reference'] = '{}/{}'.format(self.literature_base_url, self.pmid) if self.pmid else '{}/0000'.format(self.literature_base_url)
+
+        return return_value
+
+    def generate_provenance(self):
+        # Generate provenence type:
+        provenence = {
+            'expert': {
+                'status': True,
+                'statement': 'Primary submitter of the data'
+            },
+            'database': {
+                'version': self.date_added,
+                'id': self.source_id,
+                'dbxref': {
+                    'version': self.date_added,
+                    'id': self.genetics_portal_base_url
+                }
+            }
+        }
+
+        if self.literature_prop:
+            provenence['literature'] = self.literature_prop
+
+        return provenence
+
+    def generate_gene2variant(self):
+
+        # Generate evidence code field:
+        evidence_codes = [
+            self.evidence_code,
+            self.genetics_portal_base_url
+        ]
+
+        # Generate resource score field:
+        resource_score = {
+            'type': 'locus_to_gene_score',
+            'method': {
+                'description': 'Locus to gene score generated by OpenTargets Genetics portal',
+            },
+            'value': self.l2g_score
+        }
+
+        return_data = {
+            'provenance_type': self.generate_provenance(),
+            'is_associated': True,
+            'date_asserted': self.date_added,
+            'evidence_codes': evidence_codes,
+            'resource_score': resource_score,
+            'functional_consequence': self.consequence_link,
+            'consequence_code': self.consequence_code
+        }
+
+        return return_data
+
+
+def initialize_evidence_generation(df):
+    """
+    As it is not possible for multiprocessing.Pool to copy over states of objects,
+    this function instantiates the evidence generator class and generates evidences.
+    """
+
+    global schemaFile
+
+    # Fetching schema:
+    json_schema = requests.get(schemaFile).json()
+
+    # Initialize evidence builder object:
+    evidence_builder = genetics_portal_evidence_generator(json_schema, '1.6.4')
+
+    # Generate evidence for all rows of the dataframe:
+    evidences = df.apply(evidence_builder.get_evidence_string, axis=1)
+
+    return evidences
+
+
+def parallelize_dataframe(df, schema_url, n_cores=2):
+
+    # Splitting dataframe as many chunks as many cores we have defined:
+    df_split = np.array_split(df, n_cores)
+
+    # Initialize pool object:
+    pool = Pool(n_cores)
+
+    # Execute funciton on dataframe chunks and pool output:
+    df = pd.concat(pool.map(initialize_evidence_generation, df_split))
```

Comment:
Why do you need to join back the files? Can you save each df chunk straight to a file? use a [uuid 4](https://stackoverflow.com/a/534847/1197021) to append to each filename with a prefix? that would help the provision of future scalability improvements.